sumologic/resource_sumologic_cse_match_list.go

```diff
@@ -0,0 +1,307 @@
+	} else {
+		setItems(d, CSEMatchListItems.CSEMatchListItemsGetObjects)
+	}
+
+	return nil
+}
+
+func setItems(d *schema.ResourceData, items []CSEMatchListItemGet) {
+
+	var its []map[string]interface{}
+
+	for _, t := range items {
+		item := map[string]interface{}{
+			"id":          t.ID,
+			"description": t.Meta.Description,
+			"expiration":  t.Expiration,
+			"value":       t.Value,
+		}
+		its = append(its, item)
+	}
+
+	d.Set("items", its)
+
+}
+
+func resourceSumologicCSEMatchListDelete(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+	err := c.DeleteCSEMatchList(d.Id())
+	return err
+
+}
+
+func resourceSumologicCSEMatchListCreate(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	if d.Id() == "" {
+		id, err := c.CreateCSEMatchList(CSEMatchListPost{
+			Active:       true,
+			DefaultTtl:   d.Get("default_ttl").(int),
+			Description:  d.Get("description").(string),
+			Name:         d.Get("name").(string),
+			TargetColumn: d.Get("target_column").(string),
+		})
+
+		if err != nil {
+			return err
+		}
+		d.SetId(id)
+
+		//Match list items
+		itemsData := d.Get("items").([]interface{})
+		var items []CSEMatchListItemPost
+		for _, data := range itemsData {
+			item, _ := resourceToCSEMatchListItem([]interface{}{data})
+			items = append(items, item)
+
+		}
+
+		if len(items) > 0 {
+			err2 := c.CreateCSEMatchListItems(items, id)
+			if err2 != nil {
+				log.Printf("[WARN] An error occurred while adding match list items to match list id: %s, err: %v", id, err2)
+			}
+
+			// Calling Sleep method, adding items might take a while before items retrieved in next section
+			// Need to find a better way since feels super hacky
+			time.Sleep(15 * time.Second)
+
+		}
+
+	}
+
+	return resourceSumologicCSEMatchListRead(d, meta)
+}
+
+func resourceToCSEMatchListItem(data interface{}) (CSEMatchListItemPost, string) {
+	itemsSlice := data.([]interface{})
+	item := CSEMatchListItemPost{}
+	if len(itemsSlice) > 0 {
+		itemObj := itemsSlice[0].(map[string]interface{})
+		item.ID = itemObj["id"].(string)
+		item.Description = itemObj["description"].(string)
+		item.Active = true
+		item.Expiration = itemObj["expiration"].(string)
+		item.Value = itemObj["value"].(string)
+	}
+	return item, item.ID
+}
+
+func resourceSumologicCSEMatchListUpdate(d *schema.ResourceData, meta interface{}) error {
+	CSEMatchListPost, err := resourceToCSEMatchList(d)
+	if err != nil {
+		return err
+	}
+
+	c := meta.(*Client)
+	if err = c.UpdateCSEMatchList(CSEMatchListPost); err != nil {
+		return err
+	}
+
+	//Match list items
+	itemsData := d.Get("items").([]interface{})
+	var itemIds []string
+	var items []CSEMatchListItemPost
+	for _, data := range itemsData {
+		item, id := resourceToCSEMatchListItem([]interface{}{data})
+		items = append(items, item)
+		itemIds = append(itemIds, id)
+
+	}
+
+	if len(items) > 0 {
+		for _, item := range items {
+			CSEMatchListItem, er := c.GetCSEMatchListItem(item.ID)
```

Comment:
You should be able to do this in a single `c.CreateCSEMatchListItems(items, id)` API call, rather than doing multiple API calls per item like you're doing here.

Reply:
actually that is what I am doing when creating resource; but that part where you put the comment is for "updates"; so guess what I tried to do there is know if and update or creation was needed for a given list item. In a nutshell if it exists then do update on list item, if not create the list item; makes sense?

Reply:
I see what you mean now, thanks, done under https://github.com/SumoLogic/terraform-provider-sumologic/pull/353/commits/ae7e712cfc1bb072b58631fab14948da9e99413b